aerotech_101smc2/aerotech_101smc2/_aerotech.py

```diff
@@ -70,30 +70,11 @@ async def update_state(self):
             self._serial_port.write(b"Q\n")
             line = await self._serial_port.areadline()
             self._busy = (line[0] != b"R")
-            #print(line[0])
+            self.logger.debug(line[0])
             if self._busy:
                 await asyncio.sleep(0.1)
-            else:
-                await self._busy_sig.wait()
                 self._serial_port.write(b"G\n")
                 self._position = float(await self._serial_port.areadline())
```

Comment:
I would put these two lines in the main block, not inside the if block... It doesn't matter too much, the main difference is whether you wait 100 msec later

Reply:
Even putting here may not be enough as on occasion the stage may be stopping right between reading port and checking for busy() on the next index.  Suggest we have a position report in else statement before awaiting the busy_sig as well as the above